site/app/libraries/database/DatabaseQueries.php

```diff
@@ -1252,7 +1252,19 @@ public function getGradeableVersionHasAutogradingResults($g_id, $version, $user_
         return count($this->course_db->rows()) > 0 && $this->course_db->row()['autograding_complete'] === true;
     }
 
+    /**
+     * Create a string that consists placeholder parameters (?) enclosed in parentheses.
+     * These placeholders will then be used for parameter binding when executing the SQL query.
+     * When $len is zero, this function will return '(NULL)' so that it won't break SQL syntax.
+     *
+     * @param integer $len the number of placeholder parameters.
+     *
+     * @return string
+     */
     protected function createParameterList($len) {
```

Comment:
We want to make all new types also use native PHP syntax
```suggestion
     */
    protected function createParameterList(int $len): string {
```